### PR TITLE
Fixes #23120: Refactor TechniqueReader code 

### PR DESCRIPTION
**Pull request by fanf**
https://issues.rudder.io/issues/23120

Pure refactoring. It does: 
- bring back `TechniqueReader` along with other technique editor things like `TechniqueWriter`, etc
- switch the parsing of `generic_methods.json` to zio json and add a test that it at least knows how to read the currect 7.3 (rudder base + dsc) generic methods
- make config parameters for rudderc bin (use the one already existing) and generic method system & local directories

### CODE REVIEW

**Comment by fanf** [comment]
PR updated with a new commit

File: webapp/sources/rudder/rudder-web/src/main/scala/bootstrap/liftweb/RudderConfig.scala [hunk 1]
```diff
@@ -1004,6 +1004,24 @@ object RudderParsedProperties {
     }
   }
 
+  // Comes with the rudder-server packages
+  val GENERIC_METHODS_SYSTEM_LIB = {
+    try {
+      config.getString("")
```
Comment (amousset) [inline webapp/sources/rudder/rudder-web/src/main/scala/bootstrap/liftweb/RudderConfig.scala:1010]: missing key

Comment (amousset) [inline webapp/sources/rudder/rudder-web/src/main/scala/bootstrap/liftweb/RudderConfig.scala:1010]:
```suggestion
      config.getString("rudder.technique.methods.systemLib")
```

**Comment by amousset** [comment]
OK, squash merging this PR

**Pull request merged**